### ISSUE TITLE
pg_hba.conf

### ISSUE THREAD

**Issue by SamyOteroGlez** [kartoza/docker-postgis#265]
Hello team,

I am deploying this image into a docker swarm, using a custom pg_hba.conf file I created because I need to delegate the authentication to an LDAP server. 

In the version 9.4-2.1, the deployment occurs as expected, and my custom pg_hba.conf file gets added to the container defining a new configuration file (using Configs from docker swarm) thru the docker swarm (basically the file gets read-only permissions in the container if added thru Configs).

In the following versions (specifically 11.5-2.5), this process cant be accomplished due to an error that occurs in setup-pg_hba.sh line 13 (cat ${ROOT_CONF}/pg_hba.conf.template > ${ROOT_CONF}/pg_hba.conf).

So, basically I cant define in the pg_hba.conf file any other configuration other than the defaults.

If I am missing any ENVIRONMENT vars to add a custom pg_hba.conf file to add custom lines to the pg_hba.conf file please point me in the right direction.

Cheers

**Comment by NyakudyaA** [comment]
Maybe you need to add an entrypoint bash script that creates the lock file that prevents initialization of the ph-hba and then mount your config file

**Comment by lucernae** [comment]
Just to add. Mount your own `setup-pg_hba.sh` to replace the default one, and make it do necessary preparations according to your SOP (remove the line that tries to overwrite to pg_hba.conf because you have it via Config).

If you are using docker-swarm, I would suggest that your `setup-pg_hba.sh` script made as Config too, so you won't have to rebuild our image. We separate these entrypoint scripts into several files so people can override part of it with ease.

**Comment by NyakudyaA** [comment]
This had been addressed.
